File: Java/Tests/jdi-uitest-wintests/src/main/java/com/epam/jdi/uitests/testing/unittests/pageobjects/sections/JdiSearch.java
package com.epam.jdi.uitests.testing.unittests.pageobjects.sections;

import com.epam.jdi.uitests.core.interfaces.common.IButton;
import com.epam.jdi.uitests.core.interfaces.common.ITextField;
import com.epam.jdi.uitests.win.winium.elements.common.Button;
import com.epam.jdi.uitests.win.winium.elements.composite.Search;
import org.openqa.selenium.By;
import org.openqa.selenium.support.FindBy;

/**
 * Created by Roman_Iovlev on 12/17/2015.
 */
public class JdiSearch extends Search {
    @FindBy(css = ".search-field input")
    public ITextField searchInput;

    @FindBy(css = ".search .search-active .icon-search")
    public IButton searchButton;

    @Override
    public void findAction(String text) {
        new Button(By.cssSelector(".search>.icon-search")).click();
        super.findAction(text);
    }

}

File: C#.Net/Tests/JDI UIWebTests/JDI UIWebTests/UIObjects/Sections/JdiSearch.cs
using Epam.JDI.Core.Interfaces.Common;
using JDI_Web.Selenium.Elements.Composite;
using OpenQA.Selenium.Support.PageObjects;

namespace JDI_UIWebTests.UIObjects.Sections
{
    public sealed class JdiSearch : Search
    {
        [FindsBy(How = How.CssSelector, Using = ".search-field input")]
        public ITextField SearchInput;

        [FindsBy(How = How.CssSelector, Using = ".search>.icon-search")]
        public new IButton SearchButton;

        [FindsBy(How = How.CssSelector, Using = ".icon-search.active")]
        public IButton SearchButonActive;
    }
}
